SpringKonstant/src/org/usfirst/frc6357/robotcode/subsystems/TFMini.java

```diff
@@ -56,31 +56,41 @@
     private final int baudRate = 115200;	// The baud rate to configure the serial port
     private final int dataBits = 8;			// The number of data bits per transfer
     private final SerialPort.StopBits stopBits = SerialPort.StopBits.kOne;
-    private final SerialPort.Parity parity = SerialPort.Parity.kEven;
+    private final SerialPort.Parity parity = SerialPort.Parity.kNone;
+    private final SerialPort.Port port = SerialPort.Port.kOnboard; //Port used to plug in TFMini
     
     // This is a list of bytes for the initialization of the LiDAR
     private final byte[] config = {0x42, 0x57, 0x02, 0x00, 0x00, 0x00, 0x01, 0x06};
     
     // Data bytes
-    private double distL;					// byte 3: lower 8bits of distance
-    private double distH;					// byte 4: higher 8bits of distance
-    private double strengthL;				// byte 5: lower 8bit of strength
-    private double strengthH; 				// byte 6: higher 8bit of strength
-    private double reservedBytes;			// byte 7: reserved bytes
-    private double signalQualityDegree;		// byte 8: original signal quality degree
+    private int distance;    //Distance being read
+    private int strength;    //Strength of signal
+    private int quality;     //Quality of signal
+    private int checksum;    // byte 9: checksum
     
     private boolean startRead;		// Boolean to start or stop reading values
+    private ReceivingState state;   // State of the packet receive
+    private byte[] packet = new byte[9];    // Buffer for one packet
+    private int count = 0;
     
     /**
      * Constructor, creates an instance of Serial Port class.
      * @param - SerialPort.Port enum, pick where it is plugged into
      */
-    public TFMini(SerialPort.Port port)
+    public TFMini()
     {
         sensor = new SerialPort(baudRate, port, dataBits, parity, stopBits);
+        state = ReceivingState.WAIT_START1;
         initilize();
     }
     
+    private enum ReceivingState
+    {
```

Comment:
These are the correct states for the state machine.

Reply:
The intent of the enum and reading method that uses the switch is to read each byte one at a time and whenever an actual packet is found, then parse through it and store it in the byte[] packet. I'm not sure if there's any reason not to lengthen the buffer, but I do know that Java does not have unsigned data types at all so we can't use a uint.